src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Logging/RazorLogHubTraceProvider.cs

```diff
@@ -38,7 +40,7 @@ public async Task<TraceSource> InitializeTraceAsync(string logIdentifier, int lo
                 logName: $"{logIdentifier}.{logHubSessionId}",
                 serviceId: new ServiceMoniker($"Razor.{logIdentifier}"));
 
-            using var traceConfig = await LogHub.TraceConfiguration.CreateTraceConfigurationInstanceAsync(_serviceBroker, ownsServiceBroker: true, cancellationToken).ConfigureAwait(false);
+            using var traceConfig = await LogHub.TraceConfiguration.CreateTraceConfigurationInstanceAsync(_serviceBroker!, ownsServiceBroker: true, cancellationToken).ConfigureAwait(false);
```

Comment:
So I'm _fairly_ sure that `TryInitializeServiceBrokerAsync` can be decorated with `[MemberNotNullWhen(returnValue: true, Members = new[] { "_serviceBroker" })]` and then this exclamation mark should be unnecessary.. I think

Reply:
Ohh, I like that one!

Reply:
Hmm, I don't think the integration with the analyzer is complete for that one, I still get warnings after applying it.

Reply:
Thank you for trying :)